### PR TITLE
Run2-sim08 Enabling depth studies in Ecal crystal

### PR DESCRIPTION
**Pull request by bsunanda**
There was ambiguity of depth index in earlier version. Now if StoreRadiationLength is set true, it stores # of radiation lengths in unit of ScaleRadiationLength as depth; else if StoreLayerTimeSim is set true, it stores time in units of TimeSliceUnit; else it stores 0, 1, 2 for crystal volume, primary & secondary APD volumes

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@bsunanda , why a new value 1.0 is needed? Radiation length is a const for a given calorimeter. You mean a possibility to have an effective rad length, which will be tuned to data? 

Concerning EcalSD - it seems that the  class has some ineffective patterns, like string comparison at each step. Is this update for 2023 or for early workflows?

**Comment by bsunanda** [comment]
@civanch The length is stored in 16 bit integer and not in a float or double. People requiring a better precision can scale it by a factor for example 100 and can still fit in 16 bit integers. This is the purpose. There are some requirements for this - these are specific studies.
Regarding the string comparisons - they are all converted from string to int (pointer values) during initialization. All string comparisons are done during initialization - do you think any special action is needed for this

**Comment by civanch** [comment]
+1
@bsunanda , even comparison with integer seems not optimal. More effective is check for a G4Region, for that a separate regions. I would propose to add extra regions and read out logical volume data structures, of course,  in a separate PR. Concerning radiation length, I still am not sure why extra double parameter is added to g4SimHits with the default value 1.0. Do you foresee that it will be different?

File: SimG4CMS/Calo/test/python/runWithGun_cfg.py [hunk 1]
```diff
@@ -182,20 +185,20 @@
...
-    EventMin     = cms.untracked.int32(0),
-    EventMax     = cms.untracked.int32(0),
-    EventStep    = cms.untracked.int32(1),
-    PDGids       = cms.untracked.vint32(),
-    VerboseLevel = cms.untracked.int32(0),
-    G4Verbose    = cms.untracked.bool(True),
-    DEBUG        = cms.untracked.bool(False),
-    type      = cms.string('TrackingVerboseAction')
-))
+#process.g4SimHits.Watchers = cms.VPSet(cms.PSet(
```
Comment (davidlange6) [inline SimG4CMS/Calo/test/python/runWithGun_cfg.py]: hi @bsunanda - do we need the commented out lines?

Comment (bsunanda) [inline SimG4CMS/Calo/test/python/runWithGun_cfg.py]: That watcher is not working right now. I shall work on it later - make it work and then uncomment these lines

**Comment by davidlange6** [comment]
+1

**Pull request merged**